braid/_braid.h

```diff
@@ -75,13 +75,26 @@ void _braid_ErrorHandler(const char *filename, braid_Int line, braid_Int ierr, c
  * Re-allocation macro 
  **/
 #define _braid_TReAlloc(ptr, type, count) \
-( (type *)realloc((char *)ptr, (size_t)(sizeof(type) * (count))) )
+( (type *)realloc((braid_Byte *)ptr, (size_t)(sizeof(type) * (count))) )
 
 /** 
  * Free memory macro 
  **/
 #define _braid_TFree(ptr) \
-( free((char *)ptr), ptr = NULL )
+( free((braid_Byte *)ptr), ptr = NULL )
+
+/**
```

Comment:
@rfalgout @cadaverous-lives Hi Rob, David found that he needed to allocate his braid_Basis struct with a little more syntatic sugar.  The old way of just using sizeof(braid_Basis) was only returning the size of a pointer, while he actually needs more space to store a couple data members (correct me if I'm wrong David).  

Anyway, this seems a reasonable change to me, but I thought we should discuss it.

Reply:
I'm not sure we need the fancy Size0f macro.  Since all of this sizing is internal to braid, why not just use `sizeof(struct _braid_Vector_struct)` explicitly?  This macro assumes the `typedef` from opaque-object-name to pointer-to-struct follows a specific naming convention.  Of course, it currently does and it makes sense to continue that convention, but it's only a convention.

Take this with a grain of salt for now.  I will need to look more closely at what's happening here, because I don't fully understand it yet. :)  Stay tuned.

Reply:
My guess is that since the rest of the code generally follows the convention of using the pointer (say braid_Vector) instead of the underlying struct (say _braid_Vector_struct), then David wanted to follow that convention with this macro.  

Maybe one option would be rename it 
_braid_SizeOfBraidType(...) 
so that any future developer would know this isn't a general SizeOf function, but a SizeOf function that assumes some kind of Braid conventions.  I'm not sure either!  I'll think on it some more :-)